### ISSUE TITLE
Make some methods const.

### ISSUE THREAD

**Issue by manuthambi** [Lymia/enumset#27]
I would love to have some methods like `intersection`, `union` etc. const. Would help me avoid using `lazy_static!` or `once_cell::Lazy` at a bunch places in my code. This will probably require upping the MSRV. If you are open to this, I can send in a pull request.

**Comment by Lymia** [comment]
Yes, that'd be fine. I don't particularly want to maintain a strict MSRP, just a reasonable one that doesn't constantly push it for no good reason. I can also see if I can work on this myself.

**Comment by Lymia** [comment]
Do you have an idea of how to do this at a lower MSRV? This isn't even possible in stable right now because of `trait bounds other than `Sized` on const fn parameters are unstable`. What I can do is make an flag that requires nightly for const fns, but otherwise, not much I can do.

**Comment by manuthambi** [comment]
Bummer! I didn't look too carefully into it, and assumed that it should be possible for inherent methods on stable. 

Updating `enum_set!` would be pretty hacky, right? :( Perhaps it is better to just wait for more const stuff to be stabilized.

Thank you for looking into it!

**Comment by Lymia** [comment]
It would be, very much so. I'll still think about what I could do here, honestly...

**Comment by Lymia** [comment]
Reopening this since it's planned. Still blocked on rustc features - but may try some workarounds in the meanwhile.

**Comment by Aunmag** [comment]
Need a const support for `as_u8_truncated` and similar methods. I wanna do something like this:
```rs
const RAW_BIT_FILTER: u8 = enumset::enum_set!(MyFlag::A | MyFlag::B).as_u8_truncated();
```
